_delphi_utils_python/tests/test_signal.py

```diff
@@ -0,0 +1,38 @@
+"""Tests for delphi_utils.signal."""
+import pandas as pd
+from unittest.mock import patch
+
+from delphi_utils.signal import add_prefix, public_signal
+
+# Constants for mocking out the call to `covidcast.metadata` within `public_signal()`.
+SIGNALS = ["sig1", "sig2", "sig3"]
+SIGNALS_FRAME = pd.DataFrame(data={"signal": SIGNALS})
+
+class TestSignal:
+    """Tests for signal.py."""
+
+    @patch("covidcast.metadata")
+    def test_handle_wip_signal(self, metadata):
+        """Tests that `add_prefix()` derives work-in-progress signals."""
+        metadata.return_value = SIGNALS_FRAME
+
+        # Test wip_signal = True
+        signal_names = SIGNALS
+        signal_names = add_prefix(SIGNALS, True, prefix="wip_")
+        assert all(s.startswith("wip_") for s in signal_names)
```

Comment:
This is a refactor of existing stuff so I don't consider this blocking (could do in a later PR), but these tests would probably be easier if it was just a direct list comparison, e.g. `assert signal_names == ['wip_sig1', 'wip_sig2', 'wip_sig3']`. Maybe it wasn't done this way because things weren't mocked out so signal names changes?

Reply:
That probably was the reason.  I have refactored them into separate tests and made it clearer with a direct comparison.